lib/ansible/modules/cloud/ovh/ovh_dns.py

```diff
@@ -0,0 +1,312 @@
+    consumer_key = module.params.get('consumer_key')
+
+    return ovh.Client(
+        endpoint=endpoint,
+        application_key=application_key,
+        application_secret=application_secret,
+        consumer_key=consumer_key
+    )
+
+
+def get_domain_records(client, domain):
+    """Obtain all records for a specific domain"""
+    records = {}
+
+    # List all ids and then get info for each one
+    record_ids = client.get('/domain/zone/{}/record'.format(domain))
+
+    for record_id in record_ids:
+        info = client.get('/domain/zone/{}/record/{}'.format(domain, record_id))
+        add_record(records, info)
+
+    return records
+
+def add_record(records, info):
+    fieldtype = info['fieldType']
+    subdomain = info['subDomain']
+    targetval = info['target']
+
+    if fieldtype not in records:
+        records[fieldtype] = dict()
+    if subdomain not in records[fieldtype]:
+        records[fieldtype][subdomain] = dict()
+
+    records[fieldtype][subdomain][targetval] = info
+
+def find_record(records, name, fieldtype, targetval):
+    if fieldtype not in records:
+        return False
+    if name not in records[fieldtype]:
+        return False
+    if targetval not in records[fieldtype][name]:
+        return False
+
+    return records[fieldtype][name][targetval]
+
+def ensure_record_present(module, records, client):
+    domain    = module.params.get('domain')
+    name      = module.params.get('name')
+    fieldtype = module.params.get('type')
+    targetval = module.params.get('value')
+    ttl       = int(module.params.get('ttl'))
+    record    = find_record(records, name, fieldtype, targetval)
+
+    # Does the record exist already?
+    if record:
+        # The record is already as requested, no need to change anything
+        if ttl == record['ttl']:
+            module.exit_json(changed=False)
+
+        if module.check_mode:
+            module.exit_json(changed=True, diff=dict(ttl=ttl))
+
+        try:
+            # Delete and re-create the record
+            client.delete('/domain/zone/{}/record/{}'.format(domain, record['id']))
+            client.post('/domain/zone/{}/record'.format(domain), fieldType=fieldtype, subDomain=name, target=targetval, ttl=ttl)
+            client.post('/domain/zone/{}/refresh'.format(domain))
+        except APIError as error:
+            module.fail_json(
+                msg='Unable to call OVH api for recreating the record "{0} {1} {2}". '
+                'Error returned by OVH api is: "{3}".'.format(name, fieldtype, targetval, error)
+            )
+
+        module.exit_json(changed=True)
+
+    if module.check_mode:
+        module.exit_json(changed=True, diff=dict(name=name, type=fieldtype, value=targetval, ttl=ttl))
+
+    try:
+        # Add the record
+        client.post('/domain/zone/{}/record'.format(domain), fieldType=fieldtype, subDomain=name, target=targetval, ttl=ttl)
+        client.post('/domain/zone/{}/refresh'.format(domain))
```

Comment:
You should catch the error twice to produce an error message suitable for each "post"